### ISSUE TITLE
NPE happens after code analyzer fails to build a project

### ISSUE THREAD

**Issue by fabio-franco** [IBM/codenet-minerva-code-analyzer#61]
Since the code analyzer depends on a "good" build, in case of a build failure, it could fail right after the build failure and indicate to the end user to fix the app before continuing. This approach might help to avoid some NPEs we have seen in some cases. E.g.:

```
[36m2024-10-29T13:30:27.143752700[0m[35m	[INFO]	[0m[INFO] WebProductCatalogLite 0.0.1-SNAPSHOT ............... SKIPPED\
[36m2024-10-29T13:30:27.144779100[0m[35m	[INFO]	[0m[INFO] WebSessionClient1Lite 0.0.1-SNAPSHOT ............... SKIPPED\
[36m2024-10-29T13:30:27.144779100[0m[35m	[INFO]	[0m[INFO] WebSessionClient2Lite 0.0.1-SNAPSHOT ............... SKIPPED\
[36m2024-10-29T13:30:27.145763300[0m[35m	[INFO]	[0m[INFO] WebSessionClient3Lite 0.0.1-SNAPSHOT ............... SKIPPED\
[36m2024-10-29T13:30:27.145763300[0m[35m	[INFO]	[0m[INFO] WebStatefulEJBLite 0.0.1-SNAPSHOT .................. SKIPPED\
[36m2024-10-29T13:30:27.145763300[0m[35m	[INFO]	[0m[INFO] WebStockLevelLite 0.0.1-SNAPSHOT ................... SKIPPED\
[36m2024-10-29T13:30:27.145763300[0m[35m	[INFO]	[0m[INFO] WebTableLoaderLite 0.0.1-SNAPSHOT .................. SKIPPED\
[36m2024-10-29T13:30:27.145763300[0m[35m	[INFO]	[0m[INFO] ERWW_Lite_WS_Client_EAR 0.0.1-SNAPSHOT ............. SKIPPED\
[36m2024-10-29T13:30:27.145763300[0m[35m	[INFO]	[0m[INFO] ERWW_Lite_EAR 0.0.1-SNAPSHOT ....................... SKIPPED\
[36m2024-10-29T13:30:27.146786200[0m[35m	[INFO]	[0m[INFO] ERWWLiteRuntimeUtil 0.0.1-SNAPSHOT ................. SKIPPED\
[36m2024-10-29T13:30:27.146786200[0m[35m	[INFO]	[0m[INFO] ------------------------------------------------------------------------\
[36m2024-10-29T13:30:27.146786200[0m[35m	[INFO]	[0m[INFO] BUILD FAILURE\
[36m2024-10-29T13:30:27.146786200[0m[35m	[INFO]	[0m[INFO] ------------------------------------------------------------------------\
[36m2024-10-29T13:30:27.147765900[0m[35m	[INFO]	[0m[INFO] Total time:  24.573 s\
[36m2024-10-29T13:30:27.162062600[0m[35m	[INFO]	[0m[INFO] Finished at: 2024-10-29T13:30:27-07:00\
[36m2024-10-29T13:30:27.165084400[0m[35m	[INFO]	[0m[INFO] ------------------------------------------------------------------------\
[36m2024-10-29T13:30:27.183614300[0m[35m	[INFO]	[0m[ERROR] Failed to execute goal on project WebCustomSalesQuoteLite: Could not resolve dependencies for project com.ibm.ws.svt.erww-lite-ee7:WebCustomSalesQuoteLite:war:0.0.1-SNAPSHOT\
[36m2024-10-29T13:30:27.183614300[0m[35m	[INFO]	[0m[ERROR] dependency: com.ibm.ws.svt.erww-lite-ee7:WebCustomSalesQuoteLite_DownloadImageFragment:jar:0.0.1-SNAPSHOT (compile)\
[36m2024-10-29T13:30:27.184611700[0m[35m	[INFO]	[0m[ERROR] 	Could not find artifact com.ibm.ws.svt.erww-lite-ee7:WebCustomSalesQuoteLite_DownloadImageFragment:jar:0.0.1-SNAPSHOT\
[36m2024-10-29T13:30:27.184611700[0m[35m	[INFO]	[0m[ERROR] dependency: com.ibm.ws.svt.erww-lite-ee7:WebCustomSalesQuoteLite_GetImageUtilityFragment:jar:0.0.1-SNAPSHOT (compile)\
[36m2024-10-29T13:30:27.184611700[0m[35m	[INFO]	[0m[ERROR] 	Could not find artifact com.ibm.ws.svt.erww-lite-ee7:WebCustomSalesQuoteLite_GetImageUtilityFragment:jar:0.0.1-SNAPSHOT\
[36m2024-10-29T13:30:27.184611700[0m[35m	[INFO]	[0m[ERROR] dependency: com.ibm.ws.svt.erww-lite-ee7:WebCustomSalesQuoteLite_UploadImageFragment:jar:0.0.1-SNAPSHOT (compile)\
[36m2024-10-29T13:30:27.185754400[0m[35m	[INFO]	[0m[ERROR] 	Could not find artifact com.ibm.ws.svt.erww-lite-ee7:WebCustomSalesQuoteLite_UploadImageFragment:jar:0.0.1-SNAPSHOT\
[36m2024-10-29T13:30:27.186806600[0m[35m	[INFO]	[0m[ERROR] dependency: com.ibm.ws.svt.erww-lite-ee7:WebImageServingFragment:jar:0.0.1-SNAPSHOT (compile)\
[36m2024-10-29T13:30:27.186806600[0m[35m	[INFO]	[0m[ERROR] 	Could not find artifact com.ibm.ws.svt.erww-lite-ee7:WebImageServingFragment:jar:0.0.1-SNAPSHOT\
....
(omitted)
...
[36m2024-10-29T13:32:20.067718200[0m[35m	[INFO]	[0mCreate analysis scope.\
[36m2024-10-29T13:32:20.198611600[0m[35m	[INFO]	[0mAdd exclusions to scope.\
[36m2024-10-29T13:32:20.201648200[0m[35m	[INFO]	[0mLoading Java SE standard libs.\
[36m2024-10-29T13:32:20.349421[0m[33m	[WARN]	[0mNo extra libraries to process.\
[36m2024-10-29T13:32:20.401316100[0m[35m	[INFO]	[0mFinding *.class files in c:\\Users\\Administrator\\vsCode\\erww-lite\
[36m2024-10-29T13:32:23.224784700[0m[33m	[DEBUG]	[0mApplication class files: 717\
[36m2024-10-29T13:32:23.224784700[0m[35m	[INFO]	[0mAdding application classes to scope.\
[36m2024-10-29T13:32:30.111745800[0m[32m	[DONE]	[0mThere were a total of 26692 classes of which 360 are application classes.\
[36m2024-10-29T13:32:30.281755400[0m[35m	[INFO]	[0mRegistered 3626 entrypoints.\
[36m2024-10-29T13:32:30.300965400[0m[35m	[INFO]	[0mBuilding call graph.\
[36m2024-10-29T13:34:31.781118700[0m[32m	[DONE]	[0mFinished construction of call graph. Took 122.0 seconds.\
java.lang.NullPointerException: Cannot invoke "com.ibm.wala.ssa.IR.getInstructions()" because "ir" is null\
	at com.ibm.northstar.utils.AnalysisUtils.getCyclomaticComplexity(AnalysisUtils.java:89)\
	at com.ibm.northstar.SystemDependencyGraph.lambda$construct$11(SystemDependencyGraph.java:249)\
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)\
	at com.ibm.northstar.SystemDependencyGraph.construct(SystemDependencyGraph.java:246)\
	at com.ibm.northstar.CodeAnalyzer.analyze(CodeAnalyzer.java:169)\
	at com.ibm.northstar.CodeAnalyzer.run(CodeAnalyzer.java:94)\
	at picocli.CommandLine.executeUserObject(CommandLine.java:1767)\
	at picocli.CommandLine.access$900(CommandLine.java:145)\
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2139)\
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2106)\
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1973)\
	at picocli.CommandLine.execute(CommandLine.java:1902)\
	at com.ibm.northstar.CodeAnalyzer.main(CodeAnalyzer.java:85)\
```

**Comment by rofrano** [comment]
@fabio-franco It sounds like there are two things that need fixing:

1. If a build that Code Analyzer initiates fails, it should not continue but rather return a friendly error message that the build needs fixing first
2. Even so, Code Analyzer should check if the "ir" is `null` before invoking "com.ibm.wala.ssa.IR.getInstructions()", and output a friendly message instead of throwing a `NullPointerException` which is _never_ the correct thing to do.

**Comment by cherylking** [comment]
Note that this is considered a "must fix" for GA. Please update with an outlook @rofrano. Thanks.

**Comment by rofrano** [comment]
@cherylking The outlook is now: [v1.0.2](https://github.com/IBM/codenet-minerva-code-analyzer/releases/tag/v1.0.2)

**Comment by rahlk** [comment]
Fixed with version 1.0.3+
